docs/spec/fastsync/fastsync.tla

```diff
@@ -148,49 +146,82 @@ VARIABLES
 (* the variables of the node *)
 nvars == <<state, blockPool>>
 
+(*************** Type definitions for Apalache (model checker) **********************)
+AsIntSet(S) == S <: {Int}
+
+\* type of process ids
+PIDT == STRING
+AsPidSet(S) == S <: {PIDT}
+
+\* ControlMessage type
+CMT == [type |-> STRING, peerId |-> PIDT] \* type of control messages
+
+\* InMsg type
+IMT == [type |-> STRING, peerId |-> PIDT, height |-> Int, block |-> BT]
+AsInMsg(m) == m <: IMT
+AsInMsgSet(S) == S <: {IMT}
+
+\* OutMsg type
+OMT == [type |-> STRING, peerId |-> PIDT, height |-> Int]
+AsOutMsg(m) == m <: OMT
+AsOutMsgSet(S) == S <: {OMT}
+
+\* block pool type
+BPT == [height |-> Int, peerIds |-> {PIDT}, peerHeights |-> [PIDT -> Int],
+        blockStore |-> [Int -> BT], receivedBlocks |-> [Int -> PIDT],
+        pendingBlocks |-> [Int -> PIDT], syncedBlocks |-> Int, syncHeight |-> Int]
+
+AsBlockPool(bp) == bp <: BPT
+
+(******************** Sets of messages ********************************)
+
 \* Control messages
 ControlMsgs ==
-    [type: {"addPeer"}, peerId: AllPeerIds]
+    AsInMsgSet([type: {"addPeer"}, peerId: AllPeerIds])
         \union
-    [type: {"removePeer"}, peerId: AllPeerIds]
+    AsInMsgSet([type: {"removePeer"}, peerId: AllPeerIds])
         \union
-    [type: {"syncTimeout"}]
+    AsInMsgSet([type: {"syncTimeout"}])
 
 \* All messages (and events) received by a node
 InMsgs ==
-    {NoMsg}
+    AsInMsgSet({NoMsg})
         \union
-    [type: {"blockResponse"}, peerId: AllPeerIds, block: Blocks]
+    AsInMsgSet([type: {"blockResponse"}, peerId: AllPeerIds, block: Blocks])
         \union
-    [type: {"noBlockResponse"}, peerId: AllPeerIds, height: Heights]
+    AsInMsgSet([type: {"noBlockResponse"}, peerId: AllPeerIds, height: Heights])
         \union    
-    [type: {"statusResponse"}, peerId: AllPeerIds, height: Heights]
+    AsInMsgSet([type: {"statusResponse"}, peerId: AllPeerIds, height: Heights])
         \union
     ControlMsgs
 
 \* Messages sent by a node and received by peers (environment in our case)
 OutMsgs ==
-    {NoMsg}
+    AsOutMsgSet({NoMsg})
         \union
-    [type: {"statusRequest"}]           \* StatusRequest is broadcast to the set of connected peers.
+    AsOutMsgSet([type: {"statusRequest"}]) \* StatusRequest is broadcast to the set of connected peers.
         \union
-    [type: {"blockRequest"}, peerId: AllPeerIds, height: Heights]
+    AsOutMsgSet([type: {"blockRequest"}, peerId: AllPeerIds, height: Heights])
 
 
 (********************************** NODE ***********************************)
 
 InitNode ==
-     \E pIds \in SUBSET AllPeerIds \ {{}}:                   \* set of peers node established initial connections with
-        /\ blockPool = [
-                height |-> 1,
+     \E pIds \in SUBSET AllPeerIds:                   \* set of peers node established initial connections with
+        /\ pIds \subseteq CORRECT
```

Comment:
So, no faulty peers initially? Do we need that?

Reply:
A faulty peer can be added later, by executing `AddPeer`. However, I agree, we could have faulty peers from the start.